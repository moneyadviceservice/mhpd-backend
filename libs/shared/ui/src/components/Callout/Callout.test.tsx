import React from 'react';
import { render, screen } from '@testing-library/react';
import { Callout, CalloutVariant } from '.';

describe('Callout component', () => {
  it('renders the default variant correctly', () => {
    render(<Callout testId="test">Callout</Callout>);
    const container = screen.getByTestId('callout-default-test');
    expect(container).toMatchSnapshot();
  });

  it('renders the positive variant correctly', () => {
    render(
      <Callout variant={CalloutVariant.POSITIVE} testId="test">
        Callout
      </Callout>,
    );
    const container = screen.getByTestId('callout-positive-test');
    expect(container).toMatchSnapshot();
  });

  it('renders the warning variant correctly', () => {
    render(
      <Callout variant={CalloutVariant.WARNING} testId="test">
        Callout
      </Callout>,
    );
    const container = screen.getByTestId('callout-warning-test');
    expect(container).toMatchSnapshot();
  });

  it('renders the negative variant correctly', () => {
    render(
      <Callout variant={CalloutVariant.NEGATIVE} testId="test">
        Callout
      </Callout>,
    );
    const container = screen.getByTestId('callout-negative-test');
    expect(container).toMatchSnapshot();
  });

  it('renders the information variant correctly', () => {
    render(
      <Callout variant={CalloutVariant.INFORMATION} testId="test">
        Callout
      </Callout>,
    );
    const container = screen.getByTestId('callout-information-test');
    expect(container).toMatchSnapshot();
  });

  it('renders the white variant correctly', () => {
    render(
      <Callout variant={CalloutVariant.WHITE} testId="test">
        Callout
      </Callout>,
    );
    const container = screen.getByTestId('callout-white-test');
    expect(container).toMatchSnapshot();
  });
});
