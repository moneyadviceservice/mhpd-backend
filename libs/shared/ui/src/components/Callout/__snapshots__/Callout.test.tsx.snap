// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Callout component renders the default variant correctly 1`] = `
<div
  class="relative min-h-[95px] p-6 rounded-bl-3xl before:content-[""] before:block before:h-[80px] before:w-[8px] before:absolute before:left-0 before:top-0 before:rounded-bl-[8px] bg-blue-100 before:bg-blue-800"
  data-testid="callout-default-test"
>
  Callout
</div>
`;

exports[`Callout component renders the information variant correctly 1`] = `
<div
  class="relative min-h-[95px] p-6 rounded-bl-3xl before:content-[""] before:block before:h-[80px] before:w-[8px] before:absolute before:left-0 before:top-0 before:rounded-bl-[8px] bg-gray-100 before:bg-teal-400"
  data-testid="callout-information-test"
>
  Callout
</div>
`;

exports[`Callout component renders the negative variant correctly 1`] = `
<div
  class="relative min-h-[95px] p-6 rounded-bl-3xl before:content-[""] before:block before:h-[80px] before:w-[8px] before:absolute before:left-0 before:top-0 before:rounded-bl-[8px] bg-red-100 before:bg-red-700"
  data-testid="callout-negative-test"
>
  Callout
</div>
`;

exports[`Callout component renders the positive variant correctly 1`] = `
<div
  class="relative min-h-[95px] p-6 rounded-bl-3xl before:content-[""] before:block before:h-[80px] before:w-[8px] before:absolute before:left-0 before:top-0 before:rounded-bl-[8px] bg-green-100 before:bg-green-700"
  data-testid="callout-positive-test"
>
  Callout
</div>
`;

exports[`Callout component renders the warning variant correctly 1`] = `
<div
  class="relative min-h-[95px] p-6 rounded-bl-3xl before:content-[""] before:block before:h-[80px] before:w-[8px] before:absolute before:left-0 before:top-0 before:rounded-bl-[8px] bg-yellow-150 before:bg-yellow-700"
  data-testid="callout-warning-test"
>
  Callout
</div>
`;

exports[`Callout component renders the white variant correctly 1`] = `
<div
  class="relative min-h-[95px] p-6 rounded-bl-3xl before:content-[""] before:block before:h-[80px] before:w-[8px] before:absolute before:left-0 before:top-0 before:rounded-bl-[8px] bg-white before:bg-teal-400"
  data-testid="callout-white-test"
>
  Callout
</div>
`;
